File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    times = list(map(lambda idx: (100-progresses[idx] + speeds[idx] - 1) // speeds[idx], range(len(speeds)))) + [int(1e9)]
    print(times)
    
    answer, count, bef = [], 1, times[0]
    for time in times[1:]:
        if time > bef:
            bef = time
            answer.append(count)
            count = 1
        else:
            count += 1
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int n = progresses.length;
        
        int[] times = new int[n+1];
        for (int i = 0; i < n; i++)
            times[i] = (100 - progresses[i] + speeds[i] - 1) / speeds[i];
        times[n] = (int)1e9;
        
        int[] answer = new int[n+1];
        int count = 1, bef = times[0], size = 0;
        for (int i = 1; i < n+1; i++) {
            if (times[i] > bef) {
                bef = times[i];
                answer[size] = count;
                count = 1;
                size++;
            } else {
                count++;
            }
        }
        
        
        return Arrays.copyOfRange(answer, 0, size);
    }
}